File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
const fs = require('fs')
const filePath = process.platform ==="linux" ? "/dev/stdin" : "./input.txt"
const [n,m] = fs.readFileSync(filePath).toString().trim().split(" ").map(Number)
let gcd = getGCD(n,m)
console.log(gcd)
console.log(n*m/gcd)


function getGCD(a,b) {
    while(b !=0) {
        let tmp = a % b;
        a = b;
        b = tmp;
    }
    return a;
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int gcd = getGCD(n, m);
        return new int[] {gcd, n / gcd * m};
    }
    static int getGCD(int a, int b) {
        int tmp;
        while(b != 0) {
            tmp = a % b;
            a = b;
            b = tmp;
        }
        return a;
    }
        
}